### DISCUSSION TITLE
How is configuration recieved?

### DISCUSSION THREAD

**Discussion by noprompt** [noprompt/meander#179] [Ideas]
This is a continuation of the discussion appearing in #178 between @lucywang000 and myself. Here we can discuss how configuration is managed and what can be configured. Anyone else interested is free to join the discussion.

**Comment by jimmyhmiller** [comment]
Yeah, if you did a global setting, you could break things even in dependencies, which would be less than great. Personally, since these are unsafe, I think their ugliness is a feature rather than a negative.

**Comment by noprompt** [comment]
For epsilon, I think the best we're going to get is at minimum a macro which configures the compiler per namespace.

```clj
(m/configure! {,,,})
```

(I say a macro is the minimum because it can easily support each host in a uniform way.)

Not to get too deep into implementation details, but we can maintain some state which maps namespaces to their configuration and look it up at compile time and merge it in with defaults.

**Reply by lucywang000** [reply]
+1, that would strike a good balance.
